project/npda/general_functions/rcpch_nhs_organisations.py

```diff
@@ -131,116 +135,102 @@ def get_all_nhs_organisations_affiliated_with_paediatric_diabetes_unit() -> (
+            parent_name = (organisation.get("parent") or {}).get("name")
+            primary_org_name = (organisation.get("primary_organisation") or {}).get(
+                "name"
+            )
+            if parent_name:
+                organisation_list.append(
+                    (pz_code, f"{parent_name} ({primary_org_name})")
+                )
+            else:
+                organisation_list.append((pz_code, primary_org_name))
+
+        return organisation_list
+    except HTTPError as e:
+        logger.error(f"HTTP error occurred: {e.response.text}")
+        return ERROR_RESPONSE
+    except Exception as e:
+        logger.error(f"An error occurred: {e}")
+        return ERROR_RESPONSE
+
+
+def maintain_paediatric_diabetes_unit_records_against_rcpch_nhs_organisations_API():
+    """
+    This function checks if the NHS organisations in the RCPCH dataset match the locally stored PZ codes.
+    If there are any mismatches, it logs the mismatch and updates the local database with the correct data.
+    """
+    PaediatricDiabetesUnit = apps.get_model("npda", "PaediatricDiabetesUnit")
+
+    # Get all PZ codes with their trust and primary organisation
+    pdus = get_all_pz_codes_with_their_trust_and_primary_organisation(seed=True)
+
+    # Check if the NHS organisations in the RCPCH dataset match the locally stored PZ codes
+    for pdu in pdus:
+        logger.info(f"Checking PaediatricDiabetesUnit: {pdu['pz_code']}...")
+
+        # Check if the PZ code exists in the local database
+        pdu_obj = None
+        if PaediatricDiabetesUnit.objects.filter(pz_code=pdu["pz_code"]).exists():
+            pdu_obj = PaediatricDiabetesUnit.objects.get(pz_code=pdu["pz_code"])
+
+        # update or create the PaediatricDiabetesUnit
+        parent_ods_code = (pdu.get("parent") or {}).get("ods_code")
+        parent_name = (pdu.get("parent") or {}).get("name")
+
+        try:
+            new_pdu, created = PaediatricDiabetesUnit.objects.update_or_create(
+                pz_code=pdu["pz_code"],
+                defaults={
+                    "organisation_ods_code": pdu["primary_organisation"]["ods_code"],
+                    "organisation_name": pdu["primary_organisation"]["name"],
+                    "parent_ods_code": parent_ods_code,
+                    "parent_name": parent_name,
+                },
+            )
+        except Exception as e:
+            logger.error(f"Error creating PaediatricDiabetesUnit: {e}")
+            pass
+
+        if created:
+            logger.info(
+                f"Created PaediatricDiabetesUnit: {new_pdu.pz_code} ({new_pdu.organisation_name})"
+            )
+        else:
+            if pdu_obj is not None:
+                if model_to_dict(pdu_obj) != model_to_dict(new_pdu):
+                    logger.info(
+                        f"{new_pdu.pz_code} ({new_pdu.organisation_name}) was updated."
+                    )
+                else:
+                    logger.info(
```

Comment:
I think this will be quite noisy in practice so probably one to remove before merging

Reply:
this is unlikely to be run more than weekly. it is polling the API and comparing the local list with what is there and updating as necessary. For sure can remove some of the logging. At the moment for some reason the logging is not working anyway.